pkg/config/signer_test.go

```diff
@@ -106,52 +108,43 @@ func TestNewSigner(t *testing.T) {
 }
 
 func TestECDSAPublicKeyImport(t *testing.T) {
-	t.Run("certificate does not contain valid ecdsa publicKey", func(t *testing.T) {
-		gt := NewGomegaWithT(t)
-		x509cert := &x509.Certificate{PublicKey: struct{}{}}
-		_, err := ecdsaPublicKeyImport(x509cert)
-		gt.Expect(err).To(MatchError("certificate does not contain valid ECDSA public key"))
-	})
+	t.Parallel()
+
+	gt := NewGomegaWithT(t)
+	x509cert := &x509.Certificate{PublicKey: struct{}{}}
+	_, err := ecdsaPublicKeyImport(x509cert)
+	gt.Expect(err).To(MatchError("certificate does not contain valid ECDSA public key"))
 }
 
 func TestECDSAPrivateKeyImport(t *testing.T) {
-	t.Run("nil private key", func(t *testing.T) {
-		gt := NewGomegaWithT(t)
-		_, err := ecdsaPrivateKeyImport(nil)
-		gt.Expect(err.Error()).To(ContainSubstring("invalid key type. The DER must contain an ecdsa.PrivateKey"))
-	})
+	t.Parallel()
+
+	gt := NewGomegaWithT(t)
+	_, err := ecdsaPrivateKeyImport(nil)
+	gt.Expect(err.Error()).To(ContainSubstring("invalid key type. The DER must contain an ecdsa.PrivateKey"))
 }
 
 func TestSerialize(t *testing.T) {
-	t.Run("success", func(t *testing.T) {
-		gt := NewGomegaWithT(t)
-		signer, err := NewSigner([]byte(publicKey), []byte(privateKey), "test-msp")
-		gt.Expect(err).NotTo(HaveOccurred())
+	t.Parallel()
 
-		sBytes, err := signer.Serialize()
-		gt.Expect(err).NotTo(HaveOccurred())
-		serializedIdentity := &msp.SerializedIdentity{}
-		err = proto.Unmarshal(sBytes, serializedIdentity)
-		gt.Expect(serializedIdentity.Mspid).To(Equal("test-msp"))
-	})
-}
+	publicKey, privateKey := generatePublicAndPrivateKey()
 
-func TestPublic(t *testing.T) {
-	t.Run("success", func(t *testing.T) {
-		gt := NewGomegaWithT(t)
-		signer, err := NewSigner([]byte(publicKey), []byte(privateKey), "test-msp")
-		gt.Expect(err).NotTo(HaveOccurred())
+	gt := NewGomegaWithT(t)
+	signingIdentity, err := NewSigningIdentity([]byte(publicKey), []byte(privateKey), "test-msp")
```

Comment:
See above comment and all places this is casted

Reply:
done